nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/WebClient.java

```diff
@@ -236,8 +281,11 @@ protected B removeHeader(Http.HeaderName name) {
          *
          * @return socket options
          */
-        SocketOptions channelOptions() {
-            return channelOptions;
+        protected SocketOptions channelOptions() {
+            if (channelOptions != null) {
```

Comment:
Can you do this as part of the build method instead ?

Reply:
`WebClient.Builder` does not implement build method. 
Do I need to implement it? At the moment I added to the build method of the class `io.helidon.nima.webclient.http1.Http1Client.Http1ClientBuilder` that implements this interface the line : 
> configBuilder.socketOptions(super.channelOptions());

Do I need to replace this line by the code from the commented snippet above?

Reply:
You can add something like this to `WebClient.Builder`:

```java
/**
 * Actual {@link #build()} implementation for {@link WebClient} subclasses.
 *
 * @return new client
 */
protected abstract C doBuild();

@Override
public C build() {
    // do some pre-build stuff here...
    return doBuild();
}
```

Then update the current implementations of `build` to be of `doBuild`.